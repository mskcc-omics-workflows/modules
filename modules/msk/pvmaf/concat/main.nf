process PVMAF_CONCAT {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/postprocessing_variant_calls:0.3.0':
        'ghcr.io/msk-access/postprocessing_variant_calls:0.3.0' }"

    input:
    tuple val(meta), path(maf_files) // [ id:'sample1', patient:'patient1' ], [maf_1, ... maf_n]


    output:
    tuple val(meta), path("*.maf"), emit: maf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: '-sep "tsv"'
    def prefix = task.ext.prefix != null ? "${task.ext.prefix}" : (meta.patient != null ? "${meta.patient}" : "")
    def flagFiles = maf_files.collect { "-f $it" }.join(' ')
    def output = prefix ? "${prefix}_combined.maf": 'multi_sample.maf'
    """
    pv maf concat \\
        $flagFiles \\
        --output $output \\
        $args


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pv: \$( pv --version )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: '-sep "tsv"'
    def prefix = task.ext.prefix != null ? "${task.ext.prefix}" : (meta.patient != null ? "${meta.patient}" : "")
    def flagFiles = maf_files.collect { "-f $it" }.join(' ')
    def output = prefix ? "${prefix}_combined.maf": 'multi_sample.maf'
    """
    touch $output

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pv: \$( pv --version )
    END_VERSIONS
    """
}
