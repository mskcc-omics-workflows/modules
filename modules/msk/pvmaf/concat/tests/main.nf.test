nextflow_process {

    name "Test Process PVMAF_CONCAT"
    script "../main.nf"
    process "PVMAF_CONCAT"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "pvmaf"
    tag "pvmaf/concat"

    test("chr22maf - msk") {

        when {
            process {
                """input[0] = [
                            [ id:'chr22', patient:'test'], // meta map
                            file(params.test_data_mskcc['chr22']['chr22maf'], checkIfExists: true),
                            file(params.test_data_mskcc['chr22']['chr22maf'], checkIfExists: true)
                        ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }

    test("chr22maf - msk - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                        [ id:'chr22', patient:'test'], file(params.test_data_mskcc['chr22']['chr22maf'], checkIfExists: true), file(params.test_data_mskcc['chr22']['chr22maf'], checkIfExists: true)
                        ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
