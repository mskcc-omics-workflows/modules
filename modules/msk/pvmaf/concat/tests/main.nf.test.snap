{
    "chr22maf - msk - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_combined.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c61c16c135e1cc470c79e7a579c4c76c"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_combined.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c61c16c135e1cc470c79e7a579c4c76c"
                ]
            }
        ],
        "timestamp": "2024-02-23T11:31:39.015292"
    },
    "chr22maf - msk": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_combined.maf:md5,7b36c0fda7ebb28b27ddf51b7d2a09e8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c61c16c135e1cc470c79e7a579c4c76c"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_combined.maf:md5,7b36c0fda7ebb28b27ddf51b7d2a09e8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c61c16c135e1cc470c79e7a579c4c76c"
                ]
            }
        ],
        "timestamp": "2024-02-23T11:31:28.145005"
    }
}